enhancements/network/allow-mtu-changes.md

```diff
@@ -0,0 +1,207 @@
+---
+title: allow-mtu-changes
+authors:
+  - "@juanluisvaladas"
+  - "@jcaamano"
+reviewers:
+  - "@danwinship"
+  - "@dcbw"
+  - "@knobunc"
+  - "@msherif1234"
+approvers:
+  - TBD
+creation-date: 2021-10-07
+last-updated: 2021-10-07
+status: provisional
+---
+
+# Allow MTU changes
+
+This covers adding the capability to the cluster network operator of changing
+the MTU post installation.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Customers may need to change the MTU post-installation. However these changes
+aren't trivial and may cause downtime, hence the CNO currently forbids them.
+
+We propose a procedure that will be launched on demand. These procedure will
+run pods on every node of the cluster and make the necessary changes in an
+ordered and coordinated manner with a service disruption within the least
+possible time and hopefully within TCP timeout (best effort).
+
+## Motivation
+
+While cluster administrators usually set the MTU correctly during the
+installation sometimes they need to change them afterwards for reasons such as
+changes in the underlay or because they were set incorrectly at install time.
+
+### Goals
+
+* Allow to change MTU post install on OVN Kubernetes.
+
+### Non goals
+
+* Change the MTU without service disruption.
+
+* Other safe or unsafe configuration changes.
+
+## Proposal
+
+The CNO monitors changes on the operator configuration. When it detects a MTU
+change:
+1. Set the `clusteroperator/network` conditions:
+   - Progressing: true
+   - Upgradeable: false
+2. Deploy pods on every node with `restartPolicy: Never` which are responsible
+   for validating the preconditions. If the preconditions are met the pod will
+   exit with code 0. Some of the preconditions that we will check are:
+   - For MTU changes, the new MTU is valid to apply on that node.
+3. Once all the previous pods are finished, deploy other set of pods on every
+   node that will handle the actual change of the MTU. Wait for them to be
+   ready and running.
+4. Cordon every node, we don't want pods created from this point on.
+5. Ensure via machine-config-operator, that upon reboot, configure-ovs will
+   set the br-ex and its physical port MTUs to the new MTU value.
+6. Ensure that the configmap ovnkube-config is synchronized with the new MTU
+   value.
+7. Set the new MTU value to the applied cluster configuration. At this point
+   CNO is commited to the MTU change.
+8. If any step was unsuccesful, the CNO will set the `clusteroperator/network`
+   conditions to:
```

Comment:
Does "step" here refer to the steps described here or to sub-steps of one of the items? (eg, the "steps to change the MTU" of the step 3 pods)